lib/roby/interface/client.rb

```diff
@@ -273,6 +286,27 @@ def call(path, m, *args)
 
             # @api private
             #
+            # Asynchronously call a method on the interface or on one of the
+            # interface's subcommands
+            #
+            # @param [Array<String>] path path to the subcommand. Empty means on
+            #   the interface object itself.
+            # @param [Symbol] m command or action name. Actions are always
+            #   formatted as action_name!
+            # @param [Object] args the command or action arguments
+            def async_call(path, m, *args, &block)
+                if m.to_s =~ /(.*)!$/
+                    job_id = call(path, m, *args)
+                    yield(job_id, path, m, *args) if block_given?
+                else
+                    io.write_packet([path, m, *args])
+                    pending_async_calls << {:block => block, :path => path,
```

Comment:
Use the `{ block: ... }` syntax for hashes with symbol arguments (for consistency, I'm converting all of them little by little)